doc/release/upcoming_changes/19173.change.rst

```diff
@@ -0,0 +1,6 @@
+Calling `__array__` raises `ValueError` in never copy mode
+----------------------------------------------------------
```

Comment:
This won't make sense to the reader, since never-copy mode is not a thing in the main namespace.

If this needs a release mode (which I'm not sure it does), please make clear what the message is for an end user ("ignore, doesn't concern you) and for a downstream library author ("we have private support and will use that in `numpy.array_api`, in the main namespace it _may_ be coming in the near future -see PR-xxxx for details").

@mattip can you add why you're asking for this release note / what you want to see here?

Reply:
Yeah. I didn't want to mention never copy mode in release notes as initially it will be a private thing. It will be great if reviewers can point me to the right direction that can help in making this entry right for the users.

Reply:
I guess we can do without a release note, sorry for the noise.